core/src/language/python.rs

```diff
@@ -621,6 +460,155 @@ impl Python {
+                                    generic_parameters.join(", ")
+                                )
+                                .unwrap();
+                            }
+                        }
+                        other => {
+                            let mut generics = vec![];
+                            if let RustType::Simple { id } = other {
+                                // This could be a bare generic
+                                if shared.generic_types.contains(id) {
+                                    generics = vec![id.clone()];
+                                }
+                            }
+                            if generics.is_empty() {
+                                self.add_import("pydantic".to_string(), "BaseModel".to_string());
+                                writeln!(w, "class {}(BaseModel):", variant_name).unwrap();
+                            } else {
+                                self.add_import("typing".to_string(), "Generic".to_string());
+                                self.add_import(
+                                    "pydantic.generics".to_string(),
+                                    "GenericModel".to_string(),
+                                );
+                                writeln!(
+                                    w,
+                                    "class {}(GenericModel, Generic[{}]):",
+                                    // note: generics is always unique (a single item)
+                                    variant_name,
+                                    generics.join(", ")
+                                )
+                                .unwrap();
+                            }
+                        }
+                    }
+                    let python_type = self
+                        .format_type(ty, shared.generic_types.as_slice())
+                        .unwrap();
+                    writeln!(w, "    {content_key}: {python_type}")?;
+                    writeln!(w)?;
+                }
+                RustEnumVariant::AnonymousStruct { fields, .. } => {
+                    writeln!(w, "class {}Inner(BaseModel):", class_name)?;
+                    for field in fields {
+                        self.write_field(w, field, &shared.generic_types)?;
+                    }
+                    writeln!(w)?;
+                }
+            }
+        }
+        // finally, write the enum class itself consists of a type and a union of all the enum variants
+
+        writeln!(w, "class {}(BaseModel):", shared.id.renamed)?;
+        writeln!(w, "    {tag_key}: {}Types", shared.id.renamed)?;
+        writeln!(
+            w,
+            "    {content_key}: Union[{}]",
+            variant_class_names.join(", ")
+        )?;
+        writeln!(w)?;
+        // write pydantic model config
+        writeln!(w, "    class Config:")?;
+        writeln!(w, "        use_enum_values = True")?;
+        Ok(())
```

Comment:
would this be printed out for every single enum? I thought these are only needed for tagged enums?